lib/ansible/modules/windows/win_psmodule.py

```diff
@@ -11,68 +12,116 @@
                     'status': ['preview'],
                     'supported_by': 'community'}
 
-DOCUMENTATION = r'''
+DOCUMENTATION = '''
 ---
 module: win_psmodule
 version_added: "2.4"
-short_description: Adds or removes a Powershell Module
+short_description: Adds or removes a Windows PowerShell Module
 description:
-    - This module helps to install Powershell modules and register custom modules repository on Windows Server.
+  - This module helps to install Windows PowerShell modules and register custom modules repository on Windows-based systems.
 options:
   name:
     description:
-      - Name of the powershell module that has to be installed.
-    required: yes
+      - Name of the Windows PowerShell module that has to be installed.
+    required: no
+  state:
+    description:
+      - If C(present) a new module is installed.
+      - If C(absent) a module is removed.
+      - If C(latest) a module is updated to the newest version. The option was added in version 2.8.
+    choices: [ absent, latest, present ]
+    default: present
+  required_version:
+    description:
+      - The exact version of the PowerShell module that has to be installed.
+    type: str
+    version_added: "2.8"
+  minimum_version:
+    description:
+      - The minimum version of the PowerShell module that has to be installed.
+    type: str
+    version_added: "2.8"
+  maximum_version:
+    description:
+      - The maximum version of the PowerShell module that has to be installed.
+    type: str
+    version_added: "2.8"
   allow_clobber:
     description:
-      - If C(yes) imports all commands, even if they have the same names as commands that already exists. Available only in Powershell 5.1 or higher.
+      - If C(yes) imports all commands, even if they have the same names as commands that already exists. Available only in PowerShell 5.1 or higher.
     type: bool
-    default: 'no'
+    default: no
+  skip_publisher_check:
+    description:
+      - If C(yes) allows installs a newer version of a module that already exists on your computer in the case when a newer one is not
```

Comment:
`If C(yes), allows you to install a newer version of a module that already exists on your computer in the case when a newer one is not digitally signed by a trusted publisher and the newest existing module is digitally signed by a trusted publisher.`